plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AllFilesTable.java

```diff
@@ -0,0 +1,172 @@
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slices;
+import io.trino.plugin.iceberg.util.PageListBuilder;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.FixedPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.ArrayType;
+import io.trino.spi.type.TypeManager;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.transforms.Transforms;
+import org.apache.iceberg.types.Conversions;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.TypeSignature.mapType;
+import static io.trino.spi.type.VarbinaryType.VARBINARY;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class AllFilesTable implements SystemTable {
+    private final ConnectorTableMetadata tableMetadata;
+    private final Table icebergTable;
+
+    public AllFilesTable(SchemaTableName tableName, TypeManager typeManager, Table icebergTable)
+    {
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+        this.tableMetadata = new ConnectorTableMetadata(requireNonNull(tableName, "tableName is null"),
+                ImmutableList.<ColumnMetadata>builder()
```

Comment:
I think we should add the snapshot id with every row in this table. That's a divergence from [iceberg impl](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/AllDataFilesTable.java#L34), but seems good to have. cc @alexjo2144 @findepi @electrum

Reply:
@phd3
wondering, then we might need to add to the `$files` table as well?

Reply:
$files is specific to one provided snapshot, so imo does not matters much.

Reply:
@phd3 yeah, that should be fine, if we don't add.

so, please suggest, should we add the `snapshot-id` to the `$all_files`, or ok for the next improvement work? :)